## Name : Pranav Narkhede
Assignment Week-4

  
  

Apply Bayes' Theorem to solve a medical diagnosis problem given conditional probabilities.Find the eigenvalues and corresponding eigenvectors of a given matrix.Calculate the determinant of a 3x3 matrix and find its inverse if possible.Describe the properties and applications of the Normal distribution and calculate probabilities using its properties





Problem Statement
Consider a medical diagnosis problem in which there are two alternative hypotheses: 
(1) that the patient has a particular form of cancer. 
(2) that the patient does not. 
The available data is from a particular laboratory test with two possible outcomes:  + (positive) and − (negative).
The test returns a correct positive result in only 98% of the cases in which the disease is actually present, and a correct negative result in only 97% of the cases in which the disease is not present.
Furthermore, we know that 0.008 of the entire population has cancer.

In [5]:
# Given probabilities  Positive Report==>T ,Negative Report==>F

#Probability of  having Cancer
P_C = 0.008  
print(P_C)

#Probability of  not having Cancer
P_not_C=1-P_C
print(P_not_C)

#Probability of  having Cancer and positive report   
P_C_given_T = 0.98
print(P_C_given_T)

#Probaility of Report negative and  Cancer
P_C_given_F=1-P_C_given_T
print(P_C_given_F)

#Probaility of Report negative and  no Cancer
P_F_given_not_C = 0.97 
print(P_F_given_not_C )

#Probaility of Report positive and  no Cancer
P_T_given_not_C = 1 - 0.97 
print(P_T_given_not_C )

0.008
0.992
0.98
0.020000000000000018
0.97
0.030000000000000027


Suppose a new patient is observed for whom the lab test returns a positive (+) result.Should we diagnose the patient as having cancer or not?

 




Applying Bayes Theorm
P(A|B) = P(B|A)P(A) / P(B)
In many learning scenarios, the learner considers some set of candidate hypotheses H
and is interested in finding the most probable hypothesis h ∈ H given the observed data
D. Any such maximally probable hypothesis is called a maximum a posteriori (MAP)
hypothesis. Bayes theorem to calculate the posterior probability of each candidate hypothesis is hMAP
is a MAP hypothesis provided

hmap=argmax(P(A/B))
Hence P(B) can be dropped, because it is a constant independent of A   

In [6]:
#Probability of Cancer with Positive Report
P_C_given_T= P_C_given_T * P_C
print(P_C_given_T)

#Probability of NOT Cancer with Positive Report
P_not_C_given_T= P_T_given_not_C * P_not_C
print(P_not_C_given_T)

0.00784
0.029760000000000026


Conclusion
The result P_C_given_T = 0.00784 represents the probability of having cancer (C) given a positive test result (T). 
This value, despite being small, indicates that a positive test result increases the suspicion of cancer. It suggests that out of every 10,000 people who test positive, approximately 7.84 might actually have cancer.



P_not_C_given_T (0.02976): This is the probability of NOT having cancer (not C) given a positive test result (T).This value, although small, indicates that even with a positive test result, there's still a possibility the person might not have cancer. It suggests that out of every 100 people who test positive, approximately 2.98 might actually not have cancer.

## Eigenvalues and Eigenvectors of a Given Matrix


Eigenvalues (λ): These are special scalar values associated with a square matrix. For a non-zero vector v (eigenvector), multiplying A by v results in the eigenvalue λ times v: A * v = λ * v.


Eigenvectors (v): These are non-zero vectors associated with a matrix's eigenvalues. They define the directions in which the matrix scales a vector when multiplied.

In [7]:
#Importing Libraries
import numpy as np

# Define matrix A
A = np.array([[4, 1],
              [2, 3]])

# Calculate eigenvalues and eigenvectors
eigenvalues, eigenvectors = np.linalg.eig(A)


# In Python's NumPy library, np.linalg.eig(A) is a function used to compute the eigenvalues and eigenvectors of a square matrix A.

print("Eigenvalues:")
print(eigenvalues)

print("Eigenvectors:")
print(eigenvectors)


Eigenvalues:
[5. 2.]
Eigenvectors:
[[ 0.70710678 -0.4472136 ]
 [ 0.70710678  0.89442719]]


## Determinant and Inverse of a 3x3 Matrix


In [8]:
import numpy as np

# Define matrix B
B = np.array([[1, 2, 3],
              [0, 1, 4],
              [5, 6, 0]])

# Calculate determinant
det_B = np.linalg.det(B)
print(f"Determinant of B: {det_B:.4f}")

# Check if the matrix is invertible 
if det_B != 0:
    inv_B = np.linalg.inv(B)
    print("Inverse of B:")
    print(inv_B)
else:
    print("Matrix B is not invertible")


Determinant of B: 1.0000
Inverse of B:
[[-24.  18.   5.]
 [ 20. -15.  -4.]
 [ -5.   4.   1.]]


## Properties and Applications of the Normal Distribution


The normal distribution, also known as the Gaussian distribution, is a continuous probability distribution that is commonly used to model real-valued random variables. It is characterized by two parameters: the mean μ and the standard deviation 𝜎.

Properties

Symmetry: The normal distribution is symmetric around its mean μ.

Bell-Shaped: The normal distribution has a bell-shaped curve.

Continuous: The normal distribution is continuous, meaning that it can take any value within a certain range.

Applications

Modeling Real-Valued Random Variables: The normal distribution is often used to model real-valued random variables, such as heights, weights, and IQ scores.

Quality Control: The normal distribution is used in quality control to determine the acceptable limits for a product's quality.

Finance: The normal distribution is used in finance to model stock prices and returns.

##  Calculate probabilities using its properties

In [9]:
import scipy.stats as stats

# Define mean (mu) and standard deviation (sigma)
mu = 0
sigma = 1

# Create a normal distribution object
normal_dist = stats.norm(loc=mu, scale=sigma)

# Calculate the probability of a value being less than x (e.g., x = 1.96)
x = 1.96
probability = normal_dist.cdf(x)

print(f"Probability P(X < {x}) = {probability:.4f}")

# Calculate the probability of a value being within a range (e.g., -1.96 < X < 1.96)
x1 = -1.96
x2 = 1.96
probability_range = normal_dist.cdf(x2) - normal_dist.cdf(x1)

print(f"Probability P({x1} < X < {x2}) = {probability_range:.4f}")


Probability P(X < 1.96) = 0.9750
Probability P(-1.96 < X < 1.96) = 0.9500
